rust-stream/README.md

```diff
@@ -0,0 +1,72 @@
+rust-stream
+===========
+
+This is an implementation of BabelStream in Rust.
+
+Currently, we support three CPU threading API as devices:
+
+* Plain - basic single-threaded `for` version, see [plain_stream.rs](src/plain_stream.rs)
+* [Rayon](https://github.com/rayon-rs/rayon) - Parallel with high level API,
+  see [rayon_stream.rs](src/rayon_stream.rs)
+* [Crossbeam](https://github.com/crossbeam-rs/crossbeam) - Parallel with partitions per thread,
+  see [crossbeam_stream.rs](src/crossbeam_stream.rs)
+
+In addition, this implementation also supports the following extra flags:
+
+```
+--init    Initialise each benchmark array at allocation time on the main thread
+--malloc  Use libc malloc instead of the Rust's allocator for benchmark array allocation
+--pin     Pin threads to distinct cores, this has NO effect in Rayon devices
+```
+
+There is an ongoing investigation on potential performance issues under NUMA situations. As part of
+the experiment, this implementation made use of the
+provisional [Allocator traits](https://github.com/rust-lang/rust/issues/32838) which requires rust
+unstable. We hope a NUMA aware allocator will be available once the allocator API reaches rust
+stable.
+
+### Build & Run
+
+Prerequisites:
+
+* [Rust toolchain](https://www.rust-lang.org/tools/install)
+
+Once the toolchain is installed, enable the nightly channel:
+
+```shell
+> rustup install nightly
+> rustup default nightly # optional, this sets `+nightly` automatically for cargo calls later
```

Comment:
Instead of setting nightly as a global default you could recomment `rustup override set nightly` which sets it for the current directory only